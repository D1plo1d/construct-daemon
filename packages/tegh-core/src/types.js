/* config */
export Config, { MockConfig } from './config/types/Config'
export * as AxisTypeEnum from './config/types/AxisTypeEnum'

/* jobQueue */
export * as JobHistoryTypeEnum from './jobQueue/types/JobHistoryTypeEnum'

export JobHistoryEvent from './jobQueue/types/JobHistoryEvent'
export JobQueueGraphQL from './jobQueue/types/JobQueue.graphql.js'
export Job, { MockJob } from './jobQueue/types/Job'
export JobGraphQL from './jobQueue/types/Job.graphql.js'
export JobFile, { MockJobFile } from './jobQueue/types/JobFile'

/* printer */
export PrinterGraphQL from './printer/types/Printer.graphql.js'

/* spool */
export * as PriorityEnum from './spool/types/PriorityEnum'
export * as TaskStatusEnum from './spool/types/TaskStatusEnum'
export * as ComponentTypeEnum from './config/types/ComponentTypeEnum'

export Task, { MockTask } from './spool/types/Task'
export TaskGraphQL from './spool/types/Task.graphql.js'
