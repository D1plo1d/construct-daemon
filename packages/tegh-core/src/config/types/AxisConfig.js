import { Record } from 'immutable'
import uuid from 'uuid/v4'

export const AxisConfigFactory = Record({
  id: null,
  name: null,
  feedrate: null,
})

const AxisConfig = props => (
  AxisConfigFactory({
    id: props.id || uuid(),
    ...props(),
  })
)

export default AxisConfig
