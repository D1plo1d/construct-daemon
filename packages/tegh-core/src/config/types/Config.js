import { Record, List } from 'immutable'
import uuid from 'uuid/v4'

import HostConfig from './HostConfig'
import PrinterConfig from './PrinterConfig'
import MaterialConfig from './MaterialConfig'
import developmentConfig from '../../../../../development.config'

export const ConfigRecordFactory = Record({
  id: null,
  modelVersion: 0,
  host: null,
  printer: null,
  materials: List(),
})

const Config = ({
  id = uuid(),
  modelVersion = 0,
  host = {},
  printer = {},
  materials = [],
  ...props
} = {}) => (
  ConfigRecordFactory({
    id,
    modelVersion,
    host: HostConfig(host),
    printer: PrinterConfig(printer),
    materials: List(materials).map(material => MaterialConfig(material)),
    ...props,
  })
)

export const MockConfig = ({
  host = {},
  printer = {},
  ...props
} = {}) => (
  Config({
    ...developmentConfig,
    host: {
      ...developmentConfig.host,
      ...host,
    },
    printer: {
      ...developmentConfig.printer,
      ...printer,
    },
    ...props,
  })
)

export default Config
