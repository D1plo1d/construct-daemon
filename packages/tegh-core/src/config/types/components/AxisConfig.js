import { Record, Map } from 'immutable'
import uuid from 'uuid/v4'

export const AxisConfigFactory = Record({
  id: null,
  address: null,
  name: null,
  feedrate: null,
  extendedConfig: Map(),
})

const AxisConfig = ({
  id = uuid(),
  ...props
} = {}) => (
  AxisConfigFactory({
    ...props,
    id,
    extendedConfig: Map(props.extendedConfig),
  })
)

export default AxisConfig
