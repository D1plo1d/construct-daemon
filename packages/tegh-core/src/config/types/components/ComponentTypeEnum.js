import { List } from 'immutable'

export const CONTROLLER = 'CONTROLLER'
export const TOOLHEAD = 'TOOLHEAD'
export const BUILD_PLATFORM = 'BUILD_PLATFORM'
export const FAN = 'FAN'

const ComponentTypeEnum = List([
  CONTROLLER,
  TOOLHEAD,
  BUILD_PLATFORM,
  FAN,
])

export default ComponentTypeEnum
