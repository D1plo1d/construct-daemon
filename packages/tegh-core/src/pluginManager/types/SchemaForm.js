import { Record, Map } from 'immutable'

const SchemaForm = Record({
  id: null,
  schema: null,
  form: null,
  configPaths: Map({}),
})

export const NullSchemaForm = SchemaForm({
  id: 'NULL',
  schema: {},
  form: [],
})

export default SchemaForm
