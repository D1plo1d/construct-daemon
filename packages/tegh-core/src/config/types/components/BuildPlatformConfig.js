import { Record, Map } from 'immutable'
import uuid from 'uuid/v4'

import { BUILD_PLATFORM } from './ComponentTypeEnum'

export const BuildPlatformConfigFactory = Record({
  id: null,
  address: null,
  type: BUILD_PLATFORM,
  name: null,
  heater: false,
  extendedConfig: Map(),
})

const BuildPlatformConfig = ({
  id = uuid(),
  ...props
} = {}) => (
  BuildPlatformConfigFactory({
    ...props,
    id,
    extendedConfig: Map(props.extendedConfig),
  })
)

export default BuildPlatformConfig
