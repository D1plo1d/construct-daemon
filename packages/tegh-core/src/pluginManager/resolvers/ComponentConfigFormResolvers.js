import { NullSchemaForm } from '../types/SchemaForm'

const ComponentConfigFormResolvers = {
  ComponentConfigForm: {
    name: source => source.model.get('name'),
    schemaForm: (source, args, { store }) => {
      const state = store.getState()
      return state.schemaForms.get(source.type, NullSchemaForm)
    },
  },
}

export default ComponentConfigFormResolvers
