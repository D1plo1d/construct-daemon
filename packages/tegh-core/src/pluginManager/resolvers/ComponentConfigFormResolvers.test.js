import { Record, Map } from 'immutable'
import typeDefs from 'tegh-schema'

import deterministicTestSetup from '../../util/testing/deterministicTestSetup'
import snapshotTestResolvers from '../../util/testing/snapshotTestResolvers'

import { MockConfig } from '../../config/types/Config'
import { TOOLHEAD } from '../../config/types/components/ComponentTypeEnum'
import SchemaForm from '../types/SchemaForm'

import ComponentConfigFormResolvers from './ComponentConfigFormResolvers'

describe('ComponentConfigFormResolvers', () => {
  it('presents a consistent API', () => {
    deterministicTestSetup()

    const state = Record({
      schemaForms: Map({
        TOOLHEAD: SchemaForm({
          id: 'TOOLHEAD',
        }),
      }),
    })()

    snapshotTestResolvers({
      typeDefs,
      typeName: 'ComponentConfigForm',
      resolvers: ComponentConfigFormResolvers,
      rootValue: MockConfig().printer.components.find(c => c.type === TOOLHEAD),
      contextValue: {
        store: {
          getState: () => state,
        },
      },
    })
  })
})
