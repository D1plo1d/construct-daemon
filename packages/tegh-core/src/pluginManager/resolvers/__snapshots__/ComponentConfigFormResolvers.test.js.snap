// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentConfigFormResolvers presents a consistent API 1`] = `
Object {
  "_testComponentConfigForm": Object {
    "id": "heater_1",
    "model": Immutable.Map {
      "address": "e0",
      "name": "Extruder 1",
      "heater": true,
      "feedrate": 3,
      "materialID": "dkdow2",
    },
    "modelVersion": 1,
    "name": "Extruder 1",
    "schemaForm": Object {
      "id": "TOOLHEAD",
    },
    "type": "TOOLHEAD",
  },
}
`;
